src/main/java/com/conveyal/r5/analyst/TravelTimeComputer.java

```diff
@@ -81,18 +85,47 @@ public OneOriginResult computeTravelTimes() throws IOException {
         // Simultaneously we will find stations that allow access to the transit network.
         if (request.transitModes.isEmpty()) {
             // This search will use no transit.
-            //
             // When doing a non-transit walk search, we're not trying to match the behavior of egress and transfer
             // searches which use distance as the quantity to minimize (because they are precalculated and stored as distance,
             // and then converted to times by dividing by speed without regard to weights/penalties for things like stairs).
             // This does mean that walk-only results will not match the walking portion of walk+transit results.
-            sr.timeLimitSeconds = request.maxTripDurationMinutes * 60;
-            sr.streetMode = directMode;
-            sr.quantityToMinimize = StreetRouter.State.RoutingVariable.DURATION_SECONDS;
-            sr.route();
+            if (request.directModes.contains(LegMode.BICYCLE_RENT)) {
+                if (!this.network.streetLayer.bikeSharing) {
+                    LOG.warn("Bike sharing trip requested but no bike sharing stations in the " +
+                        "streetlayer");
+                    travelTimeReducer.finish();
+                    return null;
+                }
+                PointToPointQuery bikeQuery = new PointToPointQuery(network);
+                sr = bikeQuery.findBikeRentalPath(
+                    request,
+                    sr,
+                    true,
+                    false);
+
+                if (sr == null) {
+                    // Origin not found. Return an empty access times map, as is done by the other
+                    // conditions for other modes.
+                    // FIXME this is ugly. we should have a way to break out of the search early
+                    // (here and in other methods).
+                    // It causes regional analyses to be very slow when there are a large number
```

Comment:
This comment appears to be copied from somewhere else, it should be revised. The log message also refers to destination coordinates, while this is probably in fact an origin coordinate.